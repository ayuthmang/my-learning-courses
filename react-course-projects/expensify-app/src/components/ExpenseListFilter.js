import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortByAmount, sortByDate } from '../actions/filters';

const ExpenseListFilter = (props) => (
  <div>
    <input type="text" defaultValue={props.filters.text} onChange={e => {
      console.log(e.target.value);
      props.dispatch(setTextFilter(e.target.value));
    }} />
    <select
      value={props.filters.sortBy}
      onChange={e => {
        const value = e.target.value;
        if (value === 'amount') {
          props.dispatch(sortByAmount());
        } else {
          props.dispatch(sortByDate());
        }
      }}>
      <option value="date">Date</option>
      <option value="amount">Amount</option>
    </select>
  </div>
);

const mapStateToProps = (state) => {
  return {
    filters: state.filters
  }
}

export default connect(mapStateToProps)(ExpenseListFilter);
