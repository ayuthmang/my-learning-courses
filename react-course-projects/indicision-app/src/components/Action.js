import React from 'react';

const Action = (props) => (
	<div>
		<button disabled={!props.hasOptions} onClick={props.handlePick}>What should I do?</button>
	</div>
);

export default Action;