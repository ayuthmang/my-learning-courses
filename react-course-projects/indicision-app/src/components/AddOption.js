import React from 'react';

class AddOption extends React.Component {

	state = {
		error: undefined
	}

	onFormSubmit = (e) => {
		e.preventDefault();

		const option = e.target.elements.option.value.trim();
		const error = this.props.handleAddOption(option);

		this.setState(() => ({ error }));
		if (!error) {
			e.target.elements.option.value = '';
		}
	}

	render() {
		return (
			<div>
				<form onSubmit={this.onFormSubmit}>
					{this.state.error && <p>{this.state.error}</p>}
					<input type="text" name="option" />
					<button>Add Option</button>
				</form>
			</div>
		)
	}
}

export default AddOption;