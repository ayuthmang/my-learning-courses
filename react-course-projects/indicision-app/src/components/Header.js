import React from 'react';

const Header = (props) => (
	<div>
		<h1>{props.title}</h1>
		<h2>{props.subTitle}</h2>
	</div>
)

Header.defaultProps = {
	title: 'Indicision App'
};

export default Header;