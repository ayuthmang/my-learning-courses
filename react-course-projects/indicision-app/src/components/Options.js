import React from 'react';
import Option from './Option';

const Options = (props) => (
  <div>
    <button onClick={props.handleDeleteOptions}>Remove All</button>
    {props.options.length === 0 && <p>Please add an option to get started</p>}
    {props.options.map(option => {
      return <Option handleDeleteOption={props.handleDeleteOption} key={option} optionText={option} />
    })}
  </div>
)

export default Options;