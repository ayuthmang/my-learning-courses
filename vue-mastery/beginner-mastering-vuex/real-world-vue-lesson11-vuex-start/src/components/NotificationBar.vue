<template>
  <div class="notification-bar" :class="notificationTypeClass">
    <p>{{ notification.message }}</p>
  </div>
</template>

<script>
import { mapActions } from 'vuex'

export default {
  props: {
    notification: {
      type: Object,
      required: true
    }
  },
  mounted() {
    this.timeout = setTimeout(() => this.remove(this.notification), 5000)
  },
  computed: {
    notificationTypeClass() {
      return `-text-${this.notification.type}`
    }
  },
  beforeDestroy() {
    clearTimeout(this.timeout)
  },
  methods: mapActions('notification', ['remove'])
}
</script>

<style scoped>
.notification-bar {
  margin: 1em 0 1em;
}
</style>
