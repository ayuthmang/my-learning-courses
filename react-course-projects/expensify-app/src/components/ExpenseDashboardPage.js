import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilter from './ExpenseListFilter';

const ExpenseDashboardPage = () => (
  <div>
    <ExpenseListFilter />
    <ExpenseList />
  </div>
);

export default ExpenseDashboardPage;
