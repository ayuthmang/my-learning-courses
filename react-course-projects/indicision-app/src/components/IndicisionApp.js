import React from 'react';
import Header from './Header';
import Action from './Action';
import Options from './Options';
import AddOption from './AddOption';

class IndecisionApp extends React.Component {

	state = {
		options: []
	}

	componentDidMount() {
		try {
			const json = localStorage.getItem('options');
			const options = JSON.parse(json);
      if (options) {
        this.setState(() => ({ options }));
      }
		} catch (e) {
		}
	}

	componentDidUpdate(prevProps, prevState) {
		if (prevState.options.length !== this.state.options.length) {
			const json = JSON.stringify(this.state.options);
			localStorage.setItem('options', json);
		}
	}

	componentWillUnmount() {
    console.log('componentWillUnmount');
  }

	handleDeleteOptions = () => {
		this.setState(() => ({
			options: []
		}));
	}

	handleDeleteOption = (optionToRemove) => {
		this.setState((prevState) => ({
			options: prevState.options.filter((option) => option !== optionToRemove)
		}));
	}

	handleAddOption = (option) => {
		if (!option) {
			return 'Enter a valid value to add item.';
		} else if (this.state.options.indexOf(option) > -1) {
			return 'This option is already exist';
		}

		this.setState((prevState) => {
			return {
				options: prevState.options.concat(option)
			}
		});
	}

	handlePick = () => {
		const option = Math.floor(Math.random() * this.state.options.length)
		alert(this.state.options[option]);
	}

	render() {
		const subTitle = 'Put your life in the hands of a computer';

		return (
			<div>
				<Header subTitle={subTitle} />
				<Action hasOptions={this.state.options.length > 0} handlePick={this.handlePick} />
				<Options
					options={this.state.options}
					handleDeleteOptions={this.handleDeleteOptions}
					handleDeleteOption={this.handleDeleteOption} />
				<AddOption handleAddOption={this.handleAddOption} />
			</div>
		)
	}
}

export default IndecisionApp;