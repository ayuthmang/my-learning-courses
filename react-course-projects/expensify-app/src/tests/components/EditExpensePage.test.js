import React from 'react';
import { shallow } from 'enzyme';

import expenses from '../fixtures/expenses';
import { EditExpensePage } from '../../components/EditExpensePage';

let editExpense, removeExpense, history, wrapper;

beforeEach(() => {
  editExpense = jest.fn();
  removeExpense = jest.fn();
  history = { push: jest.fn() };
  wrapper = shallow(
    <EditExpensePage
      expense={expenses[0]}
      editExpense={editExpense}
      removeExpense={removeExpense}
      history={history}
    />
  );
});

it('should render EditExpensePage', () => {
  expect(wrapper).toMatchSnapshot();
});

it('should handle editExpense', () => {
  wrapper.find('ExpenseForm').prop('onSubmit')(expenses[0]);
  expect(editExpense).toHaveBeenLastCalledWith(expenses[0].id, expenses[0]);
  expect(history.push).toHaveBeenLastCalledWith('/');
});

it('should handle removeExpense', () => {
  wrapper.find('button').simulate('click');
  expect(removeExpense).toHaveBeenLastCalledWith({ id: expenses[0].id });
  expect(history.push).toHaveBeenLastCalledWith('/');
});
