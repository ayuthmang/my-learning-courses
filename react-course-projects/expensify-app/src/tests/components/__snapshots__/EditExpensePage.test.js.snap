// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EditExpensePage 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 195,
        "createdAt": 0,
        "description": "Gum",
        "id": 1,
        "note": "",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
